src/main/java/org/opentripplanner/raptor/util/IntIterators.java

```diff
@@ -117,4 +117,11 @@ public boolean hasNext() {
   public static IntIterator singleValueIterator(final int value) {
     return intIncIterator(value, value + 1);
   }
+
+  /**
+   * Return an empty iterator. All calls to {@link IntIterator#hasNext()} will return {@code false}.
+   */
+  public static IntIterator empty() {
+    return intIncIterator(0, 0);
```

Comment:
Under what circumstances is this called? I remember that creating empty iterators was relatively expensive, so can this return an immutable constant?

Reply:
You are right I should use a CONSTANT here - because it is super simple to do so. This is not a regular Iterator and the empty is not ues in a performance critical path, so I do not think the performance is a problem here. But, I will change it so it is safe and performant.

Reply:
With the constant implementation is become easier for the JIT compiler to inline the code - remove the loop.